src/sentry/integrations/message_builder.py

```diff
@@ -46,8 +47,12 @@ def build_attachment_title(obj: Group | GroupEvent) -> str:
         group = getattr(obj, "group", obj)
         if group.issue_category == GroupCategory.PERFORMANCE:
             title = GROUP_TYPE_TO_TEXT.get(group.issue_type, "Issue")
+        elif hasattr(obj, "occurrence") and obj.occurrence is not None:
+            title = obj.occurrence.issue_title
         else:
-            title = obj.title
+            event = group.get_latest_event()
```

Comment:
We probably need a ticket to make sure this returns the occurrence with the event when available

Reply:
https://github.com/getsentry/sentry/pull/42279 We'll merge this soon, so you can test this using that